### PR TITLE
New Keyboard : oktiboard 

### PR DESCRIPTION
**Pull request by oktayelipek**
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/OktiBoard/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x2014
+#define PRODUCT_ID      0x0010
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Teletubbies
+#define PRODUCT         Oktiboard
+#define DESCRIPTION     Macro and Funboard
+
```
Comment (zvecr) [inline keyboards/OktiBoard/config.h:12]:
```suggestion
```

File: keyboards/OktiBoard/config.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 0
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (zvecr) [inline keyboards/OktiBoard/config.h:32]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/OktiBoard/config.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 0
+#endif
```
Comment (zvecr) [inline keyboards/OktiBoard/config.h:29]:
```suggestion
```

File: keyboards/OktiBoard/kb.c [hunk 4]
```diff
@@ -0,0 +1 @@
+#include "kb.h"
```
Comment (zvecr) [inline keyboards/OktiBoard/kb.c:1]: This file needs to be called `keyboards/oktiboard/oktiboard.c`

File: keyboards/OktiBoard/kb.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#ifndef KB_H
```
Comment (zvecr) [inline keyboards/OktiBoard/kb.h:1]: This file needs to be called `keyboards/oktiboard/oktiboard.h`

File: keyboards/OktiBoard/kb.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+#define KB_H
```
Comment (zvecr) [inline keyboards/OktiBoard/kb.h:2]:
```suggestion
#pragma once
```

File: keyboards/OktiBoard/kb.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "quantum.h"
+
+#define LAYOUT_oktibro( \
```
Comment (zvecr) [inline keyboards/OktiBoard/kb.h:6]:
```suggestion
#define LAYOUT_ortho_4x4( \
```

File: keyboards/OktiBoard/kb.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+	K10, K11, K12, K13, \
+	K20, K21, K22, K23, \
+	K30, K31, K32, K33  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   K13 }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   K33 }  \
+}
+#endif
```
Comment (zvecr) [inline keyboards/OktiBoard/kb.h:17]:
```suggestion
```

File: keyboards/OktiBoard/rules.mk [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+
+## RGB Settings
+RGBLIGHT_ENABLE = yes
```
Comment (zvecr) [inline keyboards/OktiBoard/rules.mk:59]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/OktiBoard/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+ * |------+------+------+------|
+ * |      |AU_ON |AU_OFF|      |
+ * |------+------+------+------|
+ * |RGB_TG|RGB_VI|RGB_VD|      |
+ * |------+------+------+------|
+ * | LCK  |      |      | CHNG |
+ * `---------------------------'
+ */
+
+[_LAB] = LAYOUT_oktibro(
```
Comment (zvecr) [inline keyboards/OktiBoard/keymaps/default/keymap.c:105]:
```suggestion
[_LAB] = LAYOUT_ortho_4x4(
```

File: keyboards/OktiBoard/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+#define GRID LCTL(KC_G)                    // Open and close grids.
+#define TEXTLEFT LALT(LGUI(KC_L))           // Text Align Left
+#define TEXTRIGHT LALT(LGUI(KC_R))          // Text Align Right
+#define TEXTCENTER LALT(LGUI(KC_T))         // Text Align Center
+#define RUNLASTPLUG LALT(LALT(KC_P))        // Run last plugin
+
+
+enum oktiboard_layers {
+  _FIGMA = 0,
+  _LAB = 1,
```
Comment (zvecr) [inline keyboards/OktiBoard/keymaps/default/keymap.c:25]:
```suggestion
  _FIGMA,
  _LAB
```

File: keyboards/OktiBoard/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,111 @@
+#include "kb.h"
```
Comment (zvecr) [inline keyboards/OktiBoard/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/OktiBoard/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+            SEND_STRING(SS_LGUI(SS_LALT(SS_TAP(X_V))));
+            is_copied = false;
+        }
+        break;
+    }
+    return true;
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+  switch (biton32(state)) {
```
Comment (zvecr) [inline keyboards/OktiBoard/keymaps/default/keymap.c:60]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    switch (get_highest_layer(state)) {
```

File: keyboards/OktiBoard/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+ * |  LYR |  BF  |  SB  | SHG  |
+ * |------+------+------+------|
+ * | PLG  | TAL  | TAC  | TAR  |
+ * |------+------+------+------|
+ * | AUL  |      |  CPS | CHNG |
+ * `---------------------------'
+ */
+
+
+[_FIGMA] = LAYOUT_oktibro(
```
Comment (zvecr) [inline keyboards/OktiBoard/keymaps/default/keymap.c:86]:
```suggestion
[_FIGMA] = LAYOUT_ortho_4x4(
```

File: keyboards/OktiBoard/config.h [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
+
```
Comment (zvecr) [inline keyboards/OktiBoard/config.h:47]:
```suggestion
```

File: keyboards/OktiBoard/config.h [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+#define RGB_DI_PIN F6
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLED_NUM 16
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+
+/* Audio Settings */
+#define STARTUP_SONG SONG(STARTUP_SOUND)
```
Comment (zvecr) [inline keyboards/OktiBoard/config.h:57]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, 
* keyboard name needs to be lower case, `keyboards/oktiboard`
* missing licence headers
* no readme.md

**Review by fauxpark — CHANGES_REQUESTED** [review]
Also missing an info.json; since this looks to be a fairly standard ortho 4x4 layout, you can just grab the one from `layouts/default/ortho_4x4` and change the name/url/maintainer.

**Comment by oktayelipek** [comment]
@fauxpark and @zvecr I learned lots of things guys. Especially, thanks for your interest.

**Comment by oktayelipek** [comment]
I also made lowercase my folder name but it not reflect here.

File: keyboards/oktiboard/config.h [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x2014
```
Comment (noroadsleft) [inline keyboards/oktiboard/config.h:24]: This Vendor ID is [already in use](https://www.the-sz.com/products/usbid/index.php?v=0x2014&p=&n=).

```suggestion
#define VENDOR_ID       0x5454  // "TT" - Teletubbies
```

File: keyboards/oktiboard/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,111 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/oktiboard/keymaps/default/keymap.c:1]: Could you add a license header here, please?

Here's a sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/oktiboard/oktiboard.c [hunk 19]
```diff
@@ -0,0 +1 @@
+#include "oktiboard.h"
```
Comment (noroadsleft) [inline keyboards/oktiboard/oktiboard.c:1]: Could you add a license header here, please?

File: keyboards/oktiboard/oktiboard.h [hunk 20]
```diff
@@ -0,0 +1,15 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/oktiboard/oktiboard.h:1]: Could you add a license header here, please?

File: keyboards/oktiboard/info.json [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+    "url": "",
+    "maintainer": "qmk",
+    "width": 4,
+    "height": 4,
+    "layouts": {
+      "LAYOUT_ortho_4x4": {
+        "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":0, "y":1}, {"x":1, "y":1}, {"x":2, "y":1}, {"x":3, "y":1}, {"x":0, "y":2}, {"x":1, "y":2}, {"x":2, "y":2}, {"x":3, "y":2}, {"x":0, "y":3}, {"x":1, "y":3}, {"x":2, "y":3}, {"x":3, "y":3}]
+      },
+      "LAYOUT_numpad_4x4": {
+        "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0, "h":2}, {"x":0, "y":1}, {"x":1, "y":1}, {"x":2, "y":1}, {"x":0, "y":2}, {"x":1, "y":2}, {"x":2, "y":2}, {"x":3, "y":2, "h":2}, {"x":0, "y":3, "w":2}, {"x":2, "y":3}]
```
Comment (noroadsleft) [inline keyboards/oktiboard/info.json:12]: You need to either delete this block:

```suggestion
```

or add a `LAYOUT_numpad_4x4` macro to `oktiboard.h`.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/oktiboard/config.h (hunk 17 above)
Comment (oktayelipek) [reply keyboards/oktiboard/config.h:24]: Thanks @noroadsleft you saved the teletubbies 💯

File: keyboards/oktiboard/info.json (hunk 21 above)
Comment (oktayelipek) [reply keyboards/oktiboard/info.json:12]: I think, prefer to delete this. 👍

File: keyboards/oktiboard/info.json [hunk 22]
```diff
@@ -8,8 +8,5 @@
       "LAYOUT_ortho_4x4": {
         "layout": [{"x":0, "y":0}, {"x":1, "y":0}, {"x":2, "y":0}, {"x":3, "y":0}, {"x":0, "y":1}, {"x":1, "y":1}, {"x":2, "y":1}, {"x":3, "y":1}, {"x":0, "y":2}, {"x":1, "y":2}, {"x":2, "y":2}, {"x":3, "y":2}, {"x":0, "y":3}, {"x":1, "y":3}, {"x":2, "y":3}, {"x":3, "y":3}]
       },
```
Comment (noroadsleft) [inline keyboards/oktiboard/info.json:10]:
```suggestion
      }
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/oktiboard/info.json (hunk 22 above)
Comment (oktayelipek) [reply keyboards/oktiboard/info.json:10]: Thanks sir 🙌

Comment (noroadsleft) [reply keyboards/oktiboard/info.json:10]: The file won't work without this comma removed, because it makes the syntax.

Comment (oktayelipek) [reply keyboards/oktiboard/info.json:10]: Thanks @noroadsleft , I did 👍

**Review by noroadsleft — APPROVED** [review]

File: keyboards/oktiboard/readme.md [hunk 23]
```diff
@@ -0,0 +1,13 @@
...
+
+* Keyboard Maintainer: QMK Community
+* Hardware Supported: Pro Micro
+    * 16 NeoPixel Led
+
+Make example for this keyboard (after setting up your build environment):
+
+```sh
+make oktiboard:default        # builds and flashes the 12 
+```
```
Comment (fauxpark) [inline keyboards/oktiboard/readme.md:11]:
```suggestion
    make oktiboard:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/oktiboard/readme.md [hunk 24]
```diff
@@ -0,0 +1,11 @@
+# oktiboard
+
+* Keyboard Maintainer: QMK Community
+* Hardware Supported: Pro Micro
+    * 16 NeoPixel Led
+
+Make example for this keyboard (after setting up your build environment):
+
+make oktiboard:default
```
Comment (fauxpark) [inline keyboards/oktiboard/readme.md:9]:
```suggestion
    make oktiboard:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
PR lint is complaining about this:
```
☒ Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)
```

Please make sure all of these files are encoded in UTF-8 without the BOM, and use Unix line endings.

**Comment by oktayelipek** [comment]
I'm a bit Newby sir :) But I handled it. @fauxpark

File: keyboards/oktiboard/readme.md (hunk 24 above)
Comment (noroadsleft) [reply keyboards/oktiboard/readme.md]: Line 9 should lead with four spaces, as in fauxpark's comment.

File: keyboards/oktiboard/readme.md [hunk 25]
```diff
@@ -0,0 +1,12 @@
+# oktiboard
+
+* Keyboard Maintainer: QMK Community
+* Hardware Supported: Pro Micro
+    * 16 NeoPixel Led
```
Comment (noroadsleft) [inline keyboards/oktiboard/readme.md:5]:
```suggestion
* Hardware Supported: Pro Micro, 16 NeoPixel LED
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/oktiboard/readme.md (hunk 25 above)
Comment (zvecr) [reply keyboards/oktiboard/readme.md:5]: Also adding the availability line would be nice, as at the moment its unclear if this is a handwire.

Comment (oktayelipek) [reply keyboards/oktiboard/readme.md]: I moved to under handwired folder and updated the readme file. Handwired information added.

**Review by noroadsleft — APPROVED** [review]

File: keyboards/handwired/oktiboard/config.h [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5454  // "TT" - Teletubbies
```
Comment (Erovia) [inline keyboards/handwired/oktiboard/config.h:23]:
```suggestion
#define VENDOR_ID       0x5455  // "TT" - Teletubbies
```
0x5454 is already used by a VIA compatible board:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/satt/vision/config.h#L21

File: keyboards/oktiboard/readme.md [hunk 27]
```diff
@@ -0,0 +1,10 @@
+# oktiboard
```
Comment (Erovia) [inline keyboards/oktiboard/readme.md:1]: This should be `keyboards/handwired/oktiboard/readme.md`, isn't it?

File: keyboards/handwired/oktiboard/config.h [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
```
Comment (Erovia) [inline keyboards/handwired/oktiboard/config.h:49]:
```suggestion

#ifdef RGBLIGHT_ENABLE
```

File: keyboards/handwired/oktiboard/config.h [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+#define RGB_DI_PIN F6
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLED_NUM 16
+#define RGBLIGHT_HUE_STEP 6
+#define RGBLIGHT_SAT_STEP 6
+#define RGBLIGHT_VAL_STEP 6
```
Comment (Erovia) [inline keyboards/handwired/oktiboard/config.h:56]:
```suggestion
#define RGBLIGHT_VAL_STEP 6
#endif
```

File: keyboards/handwired/oktiboard/config.h (hunk 28 above)
Comment (Erovia) [reply keyboards/handwired/oktiboard/config.h:49]: This board can be built without RGB, so I think it'd make sense to make the RGB stuff conditional.

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/oktiboard/readme.md (hunk 27 above)
Comment (oktayelipek) [reply keyboards/oktiboard/readme.md:1]: yeap. it is

Comment (Erovia) [reply keyboards/oktiboard/readme.md:1]: Now there are 2 readmes. :)
`git rm` this.

File: keyboards/handwired/oktiboard/readme.md [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+<img width='400' src='https://img.imageupload.net/2020/10/02/IMG_1116.jpg' alt='oktiboard' type='oktiboard'>
+<img width='400' src='https://img.imageupload.net/2020/10/02/camphoto_1804928587.jpg' alt='oktiboard_wiring' />
+
+* Keyboard Maintainer: QMK Community
+* Hardware Supported: Pro Micro, 16 NeoPixel LED
+
+<b>Layers:</b>
+Figma Shortcuts,
+Keyboard settings,
+EmojiBoard
```
Comment (Erovia) [inline keyboards/handwired/oktiboard/readme.md:12]:
```suggestion
```
I'd probably move these to `keyboards/handwired/oktiboard/keymaps/default/readme.md`.

**Review by Erovia — CHANGES_REQUESTED** [review]

**Comment by oktayelipek** [comment]
I will fix all thing and again show it to you guys. I first time did a keyboard and learned too many things. I will back again. Thanks for your patience. :)

**Pull request closed without merging**